Table.union assertionerrors on different schema

```
Traceback (most recent call last):
  File "/tmp/aa0cd79aaa1a4f1ba652555c05c4bf85/pyscripts_rlCXpu.zip/gnomad_hail/slack_utils.py", line 77, in try_slack
    func(*args)
  File "/tmp/aa0cd79aaa1a4f1ba652555c05c4bf85/assign_subpops.py", line 16, in main
    pop_table = exome_pop_table.union(genome_pop_table)
  File "<decorator-gen-484>", line 2, in union
  File "/tmp/aa0cd79aaa1a4f1ba652555c05c4bf85/hail-devel-3da0e7424af0.zip/hail/typecheck/check.py", line 481, in _typecheck
    return __orig_func__(*args_, **kwargs_)
  File "/tmp/aa0cd79aaa1a4f1ba652555c05c4bf85/hail-devel-3da0e7424af0.zip/hail/table.py", line 1496, in union
    return Table(self._jt.union([table._jt for table in tables]))
  File "/usr/lib/spark/python/lib/py4j-0.10.4-src.zip/py4j/java_gateway.py", line 1133, in __call__
    answer, self.gateway_client, self.target_id, self.name)
  File "/tmp/aa0cd79aaa1a4f1ba652555c05c4bf85/hail-devel-3da0e7424af0.zip/hail/utils/java.py", line 196, in deco
    'Error summary: %s' % (deepest, full, hail.__version__, deepest)) from None
hail.utils.java.FatalError: AssertionError: assertion failed

Java stack trace:
java.lang.AssertionError: assertion failed
    at scala.Predef$.assert(Predef.scala:156)
    at is.hail.table.Table.union(Table.scala:931)
    at is.hail.table.Table.union(Table.scala:928)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
    at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
    at py4j.Gateway.invoke(Gateway.java:280)
    at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
    at py4j.commands.CallCommand.execute(CallCommand.java:79)
    at py4j.GatewayConnection.run(GatewayConnection.java:214)
    at java.lang.Thread.run(Thread.java:748)

Hail version: devel-5f23872
Error summary: AssertionError: assertion failed
```

--- a/python/hail/table.py
+++ b/python/hail/table.py
@@ -1492,7 +1492,15 @@ class Table(ExprContainer):
         :class:`.Table`
             Table with all rows from each component table.
         """
-
+        for i, ht, in enumerate(tables):
+            if not ht.row.dtype == self.row.dtype:
+                raise TypeError(f"'union': table {i} has a different row type.\n"
+                                f"  Expected:  {self.row.dtype}\n"
+                                f"  Table {i}: {ht.row.dtype}")
+            elif list(ht.key) != list(self.key):
+                raise TypeError(f"'union': table {i} has a different key."
+                                f"  Expected:  {list(self.key)}\n"
+                                f"  Table {i}: {list(ht.key)}")
         return Table(self._jt.union([table._jt for table in tables]))
 
     @typecheck_method(n=int)

